framework/helpers/BaseIpHelper.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace yii\helpers;
+
+/**
+ * Class BaseIpHelper provides concrete implementation for [[IpHelper]]
+ *
+ * Do not use BaseIpHelper, use [[IpHelper]] instead.
+ *
+ * @author Dmytro Naumenko <d.naumenko.a@gmail.com>
+ * @since 2.0.14
+ */
+class BaseIpHelper
+{
+    const IPV4 = 4;
+    const IPV6 = 6;
+    /**
+     * The length of IPv6 address in bits
+     */
+    const IPV6_ADDRESS_LENGTH = 128;
+    /**
+     * The length of IPv4 address in bits
+     */
+    const IPV4_ADDRESS_LENGTH = 32;
+
+    /**
+     * Gets the IP version. Does not perform IP address validation.
+     *
+     * @param string $ip the valid IPv4 or IPv6 address.
+     * @return int [[IPV4]] or [[IPV6]]
+     */
+    public static function getIpVersion($ip)
+    {
+        return strpos($ip, ':') === false ? self::IPV4 : self::IPV6;
+    }
+
+    /**
+     * Checks whether the IP address with the specified CIDR is in the subnet range.
+     *
+     * For example, the following code checks whether subnet `192.168.1.0/24` is in subnet `192.168.0.0/22`:
+     *
+     * ```php
+     * IpHelper::inRange('192.168.1.0', '24', '192.168.0.0/22'); // true
+     * ```
+     *
+     * In case you need to check whether a single IP address `192.168.1.21` is in the subnet `192.168.1.0/24`,
+     *
+     * ```php
+     * IpHelper::inRange('192.168.1.21', '32', '192.168.1.0/24'); // true
+     * ```
+     *
+     * @see https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing
+     *
+     * @param string $ip the valid IPv4 or IPv6 address
+     * @param int|string $cidr the CIDR of $ip
+     * @param string $range subnet in CIDR format e.g. `10.0.0.0/8` or `2001:af::/64`
+     * @return bool whether IP address with $cidr is in range
+     */
+    public static function inRange($ip, $cidr, $range)
```

Comment:
Why have separate cidr parameter for IP but expect CIDR to be inline on range?
I would expect the same logic for both.

Or am I missing somethng obvious here?

Reply:
Yeah, good point :) It was ok as internal API of IpValidator, but should be re-designed for public API